pisek/jobs/parts/solution.py

```diff
@@ -401,79 +403,47 @@ def __init__(
-            timeout=self.timeout,
-            env={"TEST_INPUT": self.input},
-        )
-        self._load_program(
-            self.solution,
-            stdin=sol_in,
-            stdout=sol_out,
-            stderr=self.sol_log_file,
-            timeout=self.timeout,
-        )
-        judge_res, sol_res = self._run_programs()
-        self._judge_run_result = judge_res
-        return sol_res
-
-    def _load_stderr(self, run_result: RunResult) -> dict[str, Any]:
-        KNOWN_METADATA = {"POINTS": float}
-        lines = run_result.raw_stderr().split("\n")
-        res = {"msg": lines[0]}
-        for line in lines[1:]:
-            if line.strip() == "":
-                continue
-            if line.count("=") != 1:
-                raise PipelineItemFailure(
-                    f"Metadata line must be in format 'KEY=value': {line}"
-                )
-            key, val = map(str.strip, line.split("="))
-            if key not in KNOWN_METADATA:
-                raise PipelineItemFailure(f"Unknown key {key}")
-
-            try:
-                res[key] = KNOWN_METADATA[key](val)
-            except ValueError:
-                raise PipelineItemFailure(f"Invalid value '{val}' for metadata {key}")
-
-        return res
-
-    def _judge(self) -> SolutionResult:
-        if self._judge_run_result.returncode == 42:
-            metadata = self._load_stderr(self._judge_run_result)
-            points = metadata.get("POINTS", 1.0)
-            if points == 1.0:
-                return SolutionResult(
-                    Verdict.ok,
-                    1.0,
-                    self._judge_run_result.raw_stderr(),
-                    self._quote_program(self._judge_run_result),
-                )
-            else:
-                return SolutionResult(
-                    Verdict.partial,
-                    points,
-                    self._judge_run_result.raw_stderr(),
-                    self._quote_program(self._judge_run_result),
-                )
-        elif self._judge_run_result.returncode == 43:
-            return SolutionResult(
-                Verdict.wrong_answer,
-                0.0,
-                self._judge_run_result.raw_stderr(),
-                self._quote_program(self._judge_run_result),
+        with tempfile.TemporaryDirectory() as fifo_dir:
+            fifo_from_solution = os.path.join(fifo_dir, "solution-to-manager")
+            fifo_to_solution = os.path.join(fifo_dir, "manager-to-solution")
+
+            os.mkfifo(fifo_from_solution)
+            os.mkfifo(fifo_to_solution)
+
+            # Open fifos to prevent blocking on future opens
+            os.open(fifo_from_solution, os.O_RDWR)
+            os.open(fifo_to_solution, os.O_RDWR)
+
+            fd_from_solution = os.open(fifo_from_solution, os.O_WRONLY)
+            fd_to_solution = os.open(fifo_to_solution, os.O_RDONLY)
+
+            points_file = self._output(self.input_name.replace(".in", ".judge"))
```

Comment:
I don't particulary like this as this puts points files in outputs folder, however that's for sepearate issue.

Reply:
See #127 for reference.